tests/v2/validation/KDM/kdm_test.go

```diff
@@ -0,0 +1,228 @@
+	k.Require().NoError(err, "Failed to create RancherClient")
+	k.cluster, err = k.client.Management.Cluster.ByID("local")
+	k.Require().NoError(err)
+	k.Require().NotEmpty(k.cluster)
+	localClusterKubeconfig, err := k.client.Management.Cluster.ActionGenerateKubeconfig(k.cluster)
+	k.Require().NoError(err)
+	c, err := clientcmd.NewClientConfigFromBytes([]byte(localClusterKubeconfig.Config))
+	k.Require().NoError(err)
+	k.config, err = c.ClientConfig()
+	k.Require().NoError(err)
+	k.clientset, err = kubernetes.NewForConfig(k.config)
+	k.Require().NoError(err)
+}
+
+func (k *KDMTestSuite) TearDownSuite() {
+	k.session.Cleanup()
+}
+
+func (k *KDMTestSuite) updateKDMurl(value string) {
+	existing, err := k.client.Steve.SteveType("management.cattle.io.setting").ByID(rkeMetadataConfig)
+	k.Require().NoError(err, "error getting existing setting")
+
+	var kdmSetting v3.Setting
+	err = stevev1.ConvertToK8sType(existing.JSONResp, &kdmSetting)
+	k.Require().NoError(err, "error converting existing setting")
+
+	kdmData := map[string]string{}
+	err = json.Unmarshal([]byte(kdmSetting.Value), &kdmData)
+	k.Require().NoError(err, "error unmarshaling existing setting")
+
+	kdmData["url"] = value
+	val, err := json.Marshal(kdmData)
+	k.Require().NoError(err, "error marshaling existing setting")
+	kdmSetting.Value = string(val)
+	_, err = k.client.Steve.SteveType("management.cattle.io.setting").Update(existing, kdmSetting)
+	k.Require().NoError(err, "error updating setting")
+}
+
+func (k *KDMTestSuite) ScaleRancherTo(desiredReplicas int32) {
+	deployment, err := k.clientset.AppsV1().Deployments(rancherNamespace).Get(context.TODO(), rancherDeployment, metav1.GetOptions{})
+	k.Require().NoError(err, "error getting rancher deployment")
+
+	if deployment.Spec.Replicas == &desiredReplicas {
+		return
+	}
+	deployment.Spec.Replicas = &desiredReplicas
+
+	deployment, err = k.clientset.AppsV1().Deployments(rancherNamespace).Update(context.TODO(), deployment, metav1.UpdateOptions{})
+	k.Require().NoError(err, "error updating rancher deployment")
+
+	// Wait for the deployment to scale up using exponential defaultBackoff
+	err = wait.ExponentialBackoff(defaultBackoff, func() (bool, error) {
+		deployment, err = k.clientset.AppsV1().Deployments(rancherNamespace).Get(context.TODO(), rancherDeployment, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("Error getting deployment: %s", err.Error())
+		}
+
+		if deployment.Status.ReadyReplicas == desiredReplicas {
+			log.Infof("Deployment %s successfully scaled to %d replicas", rancherDeployment, desiredReplicas)
+			return true, nil
+		}
+		log.Infof("Waiting for deployment %s to scale. Current replicas: %d/%d", rancherDeployment, deployment.Status.ReadyReplicas, desiredReplicas)
+		return false, nil
+	})
+	k.Require().NoError(err, "error scaling rancher deployment, timed out")
+
+	// arbitrary sleep for leader election activity to take place on the newly created replicas
```

Comment:
Any way to explicitly wait for leader election to complete instead of doing "arbitrary sleep"?

Reply:
To Sergey's point, we try to avoid using sleep's unless its absolutely necessary -- Is there any way to determine which pod has been elected as the leader? If there is like an annotation or something which denotes this, maybe we could check for that instead of this sleep?  -- If its not easily identifiable, I think we could perhaps make an exception w/ this, just because you do have a wait for the deployment, and once that's hit, you're essentially just giving a bit of buffer time for the leader election process.

Reply:
The (new) way in Rancher to determine the Leader is to inspect the lease in `cattle-provisioning-capi-system` namespace with the name `controller-leader-election-capi`. 
The problem in this context is, even if we know that this lease exists, we don't have a way to say that leader election happened and replicas are in sync now.